src/main/java/seedu/address/model/property/Price.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.model.property;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class Price {
+
+    public static final String MESSAGE_CONSTRAINTS = "Price must have the format $#####";
+    public static final String VALIDATION_REGEX = "^\\$\\d*";
+
+    private static final double THOUSAND = 1E3;
+    private static final double MILLION = 1E6;
+
+    public final String value;
+
+    /**
+     * Constructs a {@code Price}.
+     *
+     * @param price A valid price.
+     */
+    public Price(String price) {
+        requireNonNull(price);
+        checkArgument(isValidPrice(price), MESSAGE_CONSTRAINTS);
+        value = price;
+    }
+
+    public static boolean isValidPrice(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    public int getAsInt() {
+        return Integer.parseInt(value.substring(1));
+    }
+
+    @Override
```

Comment:
Maybe add a comment on what you are doing here (eg. We want to show price of property that are in the thousands as 100K and those in millions as 1M)